### PR TITLE
Using Poetry Package Manager

### PR DESCRIPTION
**Pull request by GbotemiB**
This PR replaces pip with Poetry Package manager to make reproducibility easier.

Poetry is implemented in the github actions to handle dependencies.

### CODE REVIEW

File: project-3/requirements.txt
Comment (elseagle) [inline project-3/requirements.txt:1]: you do not need this file when using poetry

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v4
+
+      # setup python with different versions
+      - name: setup python version ${{ matrix.python-version}}
```
Comment (elseagle) [inline .github/workflows/ci.yml]: why do you need different versions of python

**Review by elseagle — CHANGES_REQUESTED** [review]
Good start!

- Why do you have multiple environment builds?
- Merge the other branch into this branch, we only need one PR

File: project-3/requirements.txt
Comment (GbotemiB) [reply project-3/requirements.txt:1]: noted

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GbotemiB) [reply .github/workflows/ci.yml]: I read up it is a good idea to test for multiple os. since it is a small project I will review it

**Review by elseagle — APPROVED** [review]

**Pull request merged**